engine/rpcserver.go

```diff
@@ -3072,3 +3074,40 @@ func (s *RPCServer) GetRecentTrades(_ context.Context, r *gctrpc.GetSavedTradesR
 
 	return resp, nil
 }
+
+func checkParams(exchName string, e exchange.IBotExchange, a asset.Item, p currency.Pair) error {
+	if e == nil {
+		return fmt.Errorf("%s %w", exchName, errExchangeNotLoaded)
+	}
+	if !e.IsEnabled() {
+		return fmt.Errorf("%s %w", exchName, errExchangeDisabled)
+	}
+	if a.IsValid() {
+		b := e.GetBase()
+		if b == nil {
+			return errExchangeBaseNotFound
```

Comment:
To be in line with your other more helpful errors, how about `return fmt.Errorf("%s %w", exchName, errExchangeBaseNotFound)` ?